ui/rewards/steps/LoginStepContent.tsx

```diff
@@ -0,0 +1,145 @@
+  const router = useRouter();
+  const { connect, isConnected, address } = useWallet({ source: 'Merits' });
+  const savedRefCode = cookies.get(cookies.NAMES.REWARDS_REFERRAL_CODE);
+  const [ isRefCodeUsed, setIsRefCodeUsed ] = useBoolean(Boolean(savedRefCode));
+  const [ isLoading, setIsLoading ] = useBoolean(false);
+  const [ refCode, setRefCode ] = useState(savedRefCode || '');
+  const [ refCodeError, setRefCodeError ] = useBoolean(false);
+  const dividerColor = useColorModeValue('blackAlpha.200', 'whiteAlpha.200');
+  const { login, checkUserQuery } = useRewardsContext();
+  const profileQuery = useProfileQuery();
+
+  const handleRefCodeChange = React.useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setRefCode(event.target.value);
+  }, []);
+
+  const loginToRewardsProgram = useCallback(async() => {
+    try {
+      setRefCodeError.off();
+      setIsLoading.on();
+      const { isNewUser, invalidRefCodeError } = await login(isRefCodeUsed ? refCode : '');
+      if (invalidRefCodeError) {
+        setRefCodeError.on();
+      } else {
+        if (isNewUser) {
+          goNext(Boolean(refCode));
+        } else {
+          closeModal();
+          router.push({ pathname: '/account/rewards' }, undefined, { shallow: true });
+        }
+      }
+    } catch (error) {}
+    setIsLoading.off();
+  }, [ login, goNext, setIsLoading, router, closeModal, refCode, setRefCodeError, isRefCodeUsed ]);
+
+  useEffect(() => {
+    setRefCodeError.off();
+  }, [ refCode ]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const { start: loginToAccount } = useSignInWithWallet({
+    isAuth: Boolean(!profileQuery.isLoading && profileQuery.data?.email),
+    onSuccess: loginToRewardsProgram,
+    onError: setIsLoading.off,
+  });
+
+  const handleLogin = useCallback(async() => {
+    if (!profileQuery.isLoading && !profileQuery.data?.address_hash) {
+      setIsLoading.on();
+      loginToAccount();
+      return;
+    }
+    loginToRewardsProgram();
+  }, [ loginToAccount, loginToRewardsProgram, profileQuery, setIsLoading ]);
+
+  const isAddressMismatch =
+    Boolean(address) &&
+    Boolean(profileQuery.data?.address_hash) &&
+    profileQuery.data?.address_hash !== address;
+
+  return (
+    <>
+      <Image
+        src="/static/merits_program.png"
+        alt="Merits program"
+        mb={ 3 }
+        fallback={ <Skeleton w="full" h="120px" mb={ 3 }/> }
+      />
+      <Box mb={ 6 }>
+        Merits are awarded for a variety of different Blockscout activities. Connect a wallet to get started.
+        <LinkExternal href="https://docs.blockscout.com/using-blockscout/my-account/merits" ml={ 1 } fontWeight="500">
+          More about Blockscout Merits
+        </LinkExternal>
+      </Box>
+      { (isConnected && !isAddressMismatch && !checkUserQuery.isFetching && !checkUserQuery.data?.exists) && (
+        <Box mb={ 6 }>
+          <Box w="full" mb={ 6 } borderTop="1px solid" borderColor={ dividerColor }/>
```

Comment:
Can we use `<Divider/>` component instead? Or maybe add this border to the parent component?